Compile bug: error: use of undeclared identifier 'llama_chat_builtin_templates' building 4242

### Git commit

642330a

### Operating systems

Mac

### GGML backends

Metal

### Problem description & steps to reproduce

Trying to build using MacPorts. Previous versions build fine. Exact configuration can be found in the log below.

### First Bad Commit

I think its this one: https://github.com/ggerganov/llama.cpp/commit/642330ac7cea11dc1f9d3df2b8f3dbd10b5e3f3e#diff-f3a359c2393a8c0bf7cb0cbe9e4622c62366566bb4956ebf8d56e1bd196f8fe7R351
4231 and 4240 builds fine.

### Relevant log output

```shell
:info:build [ 32%] Linking C executable ../bin/test-c
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/tests && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/test-c.dir/link.txt --verbose=ON
:info:build /usr/bin/clang -pipe -Os -DNDEBUG -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -mmacosx-version-min=15.0 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk "CMakeFiles/test-c.dir/test-c.c.o" -o ../bin/test-c  -Wl,-rpath,/opt/local/lib ../src/libllama.dylib ../ggml/src/libggml.dylib ../ggml/src/ggml-cpu/libggml-cpu.dylib ../ggml/src/ggml-blas/libggml-blas.dylib ../ggml/src/ggml-metal/libggml-metal.dylib ../ggml/src/libggml-base.dylib
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build [ 32%] Built target test-c
:info:build [ 32%] Linking CXX executable ../../bin/llama-simple
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/examples/simple && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/llama-simple.dir/link.txt --verbose=ON
:info:build /usr/bin/clang++ -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -mmacosx-version-min=15.0 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk "CMakeFiles/llama-simple.dir/simple.cpp.o" -o ../../bin/llama-simple  -Wl,-rpath,/opt/local/lib ../../src/libllama.dylib ../../ggml/src/libggml.dylib ../../ggml/src/ggml-cpu/libggml-cpu.dylib ../../ggml/src/ggml-blas/libggml-blas.dylib ../../ggml/src/ggml-metal/libggml-metal.dylib ../../ggml/src/libggml-base.dylib
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build [ 32%] Built target llama-simple
:info:build [ 32%] Linking CXX executable ../../bin/llama-simple-chat
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/examples/simple-chat && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/llama-simple-chat.dir/link.txt --verbose=ON
:info:build /usr/bin/clang++ -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -mmacosx-version-min=15.0 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk "CMakeFiles/llama-simple-chat.dir/simple-chat.cpp.o" -o ../../bin/llama-simple-chat  -Wl,-rpath,/opt/local/lib ../../src/libllama.dylib ../../ggml/src/libggml.dylib ../../ggml/src/ggml-cpu/libggml-cpu.dylib ../../ggml/src/ggml-blas/libggml-blas.dylib ../../ggml/src/ggml-metal/libggml-metal.dylib ../../ggml/src/libggml-base.dylib
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/llama.cpp-4242/common/arg.cpp:353:19: error: use of undeclared identifier 'llama_chat_builtin_templates'
:info:build   353 |     int32_t res = llama_chat_builtin_templates(nullptr, 0);
:info:build       |                   ^
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/llama.cpp-4242/common/arg.cpp:355:11: error: use of undeclared identifier 'llama_chat_builtin_templates'
:info:build   355 |     res = llama_chat_builtin_templates(supported_tmpl.data(), supported_tmpl.size());
:info:build       |           ^
:info:build [ 32%] Built target llama-simple-chat
:info:build [ 32%] Linking CXX executable ../../bin/llama-run
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/examples/run && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/llama-run.dir/link.txt --verbose=ON
:info:build /usr/bin/clang++ -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -mmacosx-version-min=15.0 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk "CMakeFiles/llama-run.dir/run.cpp.o" -o ../../bin/llama-run  -Wl,-rpath,/opt/local/lib ../../src/libllama.dylib ../../ggml/src/libggml.dylib ../../ggml/src/ggml-cpu/libggml-cpu.dylib ../../ggml/src/ggml-blas/libggml-blas.dylib ../../ggml/src/ggml-metal/libggml-metal.dylib ../../ggml/src/libggml-base.dylib
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build [ 32%] Built target llama-run
:info:build 2 errors generated.
:info:build make[2]: *** [common/CMakeFiles/common.dir/arg.cpp.o] Error 1
:info:build make[2]: *** Waiting for unfinished jobs....
:info:build [ 32%] Linking CXX executable ../../bin/llama-quantize-stats
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/examples/quantize-stats && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/llama-quantize-stats.dir/link.txt --verbose=ON
:info:build /usr/bin/clang++ -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -mmacosx-version-min=15.0 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk "CMakeFiles/llama-quantize-stats.dir/quantize-stats.cpp.o" "../../common/CMakeFiles/build_info.dir/build-info.cpp.o" -o ../../bin/llama-quantize-stats  -Wl,-rpath,/opt/local/lib ../../src/libllama.dylib ../../ggml/src/libggml.dylib ../../ggml/src/ggml-cpu/libggml-cpu.dylib ../../ggml/src/ggml-blas/libggml-blas.dylib ../../ggml/src/ggml-metal/libggml-metal.dylib ../../ggml/src/libggml-base.dylib
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build [ 32%] Built target llama-quantize-stats
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build [ 32%] Built target llava
:info:build /Library/Developer/CommandLineTools/usr/bin/make  -f examples/llava/CMakeFiles/llava_static.dir/build.make examples/llava/CMakeFiles/llava_static.dir/depend
:info:build /Library/Developer/CommandLineTools/usr/bin/make  -f examples/llava/CMakeFiles/llava_shared.dir/build.make examples/llava/CMakeFiles/llava_shared.dir/depend
:info:build make[2]: Entering directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build && /opt/local/bin/cmake -E cmake_depends "Unix Makefiles" /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/llama.cpp-4242 /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/llama.cpp-4242/examples/llava /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/examples/llava /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/examples/llava/CMakeFiles/llava_static.dir/DependInfo.cmake "--color="
:info:build make[2]: Entering directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build && /opt/local/bin/cmake -E cmake_depends "Unix Makefiles" /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/llama.cpp-4242 /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/llama.cpp-4242/examples/llava /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/examples/llava /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build/examples/llava/CMakeFiles/llava_shared.dir/DependInfo.cmake "--color="
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build /Library/Developer/CommandLineTools/usr/bin/make  -f examples/llava/CMakeFiles/llava_shared.dir/build.make examples/llava/CMakeFiles/llava_shared.dir/build
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build /Library/Developer/CommandLineTools/usr/bin/make  -f examples/llava/CMakeFiles/llava_static.dir/build.make examples/llava/CMakeFiles/llava_static.dir/build
:info:build make[2]: Entering directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build make[2]: Entering directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_llama.cpp/llama.cpp/work/build'
:info:build [ 33%] Linking CXX static library libllava_static.a
:info:build [ 33%] Linking CXX shared library libllava_shared.dylib
```

--- a/include/llama.h
+++ b/include/llama.h
@@ -991,7 +991,7 @@ extern "C" {
                                int32_t   length);
 
     // Get list of built-in chat templates
-    int32_t llama_chat_builtin_templates(const char ** output, size_t len);
+    LLAMA_API int32_t llama_chat_builtin_templates(const char ** output, size_t len);
 
     //
     // Sampling API

